live-examples/js-examples/promise-reject.html

```diff
@@ -0,0 +1,9 @@
+<pre>
+<code id="static-js" class="language-js">Promise.reject(new Error('fail')).then(function(error) {
+  // not called
+}, function(error) {
+  console.log(error);
+});
+// expected output: Error: fail
+</code>
+</pre>
```

Comment:
I found this example pretty hard to understand. Do you think something like this might be clearer:

```
function resolved(result) {
  console.log("Resolved");
}

function rejected(result) {
  console.log("Rejected");
}

Promise.reject('fail').then(resolved, rejected);

//expected output: Rejected
```

...is this correct though?

Reply:
It is correct. I think. I shamelessly stole my example from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject where they describe encouraging the use of the Error object. The part I find confusing with reject (taking your code as an example) is that I'm not sure what we need to pass to `Promise.reject` to get `function resolved` to be called. ?

Reply:
> I'm not sure what we need to pass to Promise.reject to get function resolved to be called?

As far as I know: there is nothing you can pass. If you call `reject()`, the promise is always rejected.

Good point about Error. You could still do this:

```
function resolved(result) {
  console.log("Resolved");
}

function rejected(result) {
  console.log(result);
}

Promise.reject(new Error('fail')).then(resolved, rejected);
//expected output: "Error: fail"
```

...but if you don't think this is clearer, I'm happy to accept the existing version too.

Reply:
It seems odd that you can supply a function that will never be called. :shrug: I think your last example is a good compromise.